internal/storage/storage_integration_test.go

```diff
@@ -329,11 +331,197 @@ var _ = Describe("Storage", func() {
+					ConfigMapName:      "test",
+					ConfigMapNamespace: "default",
+				},
+			}
+			device.Spec.Storage = storageObj
+		})
+
+		It("missing S3 configuration", func() {
+			// given
+			device.Spec.Storage = &v1alpha1.Storage{}
+
+			// when
+			result, err := claimer.GetExternalStorageConfig(context.TODO(), device)
+
+			// then
+			Expect(err).ToNot(BeNil())
+			Expect(result).To(BeNil())
+		})
+
+		It("missing config map", func() {
+			// given
+			device.Spec.Storage.S3.ConfigMapName = "missing"
+			createS3Secret(nil)
+
+			// when
+			result, err := claimer.GetExternalStorageConfig(context.TODO(), device)
+
+			// then
+			Expect(err).ToNot(BeNil())
+			Expect(result).To(BeNil())
+		})
+
+		It("missing secret", func() {
+			// given
+			device.Spec.Storage.S3.SecretName = "missing"
+			createS3CM(nil)
+
+			// when
+			result, err := claimer.GetExternalStorageConfig(context.TODO(), device)
+
+			// then
+			Expect(err).ToNot(BeNil())
+			Expect(result).To(BeNil())
+		})
+
+		for fieldName, optional := range cmFields {
+			It("missing ConfigMap field "+fieldName, func() {
+				//given
+				cmData := getCmData()
+				delete(cmData, fieldName)
+				createS3CM(cmData)
+				createS3Secret(getSecretData())
+
+				// when
+				result, err := claimer.GetExternalStorageConfig(context.TODO(), device)
+
+				// then
+				if optional {
+					Expect(err).To(BeNil())
+					Expect(result).ToNot(BeNil())
+				} else {
+					Expect(err).ToNot(BeNil())
+					Expect(result).To(BeNil())
+				}
+			})
+		}
+
+		for fieldName, optional := range secretFields {
+			It("missing Secret field "+fieldName, func() {
+				//given
+				secretData := getSecretData()
+				delete(secretData, fieldName)
+				createS3CM(getCmData())
+				createS3Secret(secretData)
+
+				// when
+				result, err := claimer.GetExternalStorageConfig(context.TODO(), device)
+
+				// then
+				if optional {
+					Expect(err).To(BeNil())
+					Expect(result).ToNot(BeNil())
+				} else {
+					Expect(err).ToNot(BeNil())
+					Expect(result).To(BeNil())
+				}
+			})
+		}
+
+		It("success", func() {
+			// given
+			cmData := getCmData()
+			createS3CM(cmData)
+			secretData := getSecretData()
+			createS3Secret(secretData)
+			expectedPort, _ := strconv.Atoi(cmData["BUCKET_PORT"])
+			expected := models.S3StorageConfiguration{
+				AwsAccessKeyID:     base64.StdEncoding.EncodeToString(secretData["AWS_ACCESS_KEY_ID"]),
+				AwsSecretAccessKey: base64.StdEncoding.EncodeToString(secretData["AWS_SECRET_ACCESS_KEY"]),
+				AwsCaBundle:        base64.StdEncoding.EncodeToString(secretData["tls.crt"]),
+				BucketHost:         cmData["BUCKET_HOST"],
+				BucketPort:         int32(expectedPort),
+				BucketName:         cmData["BUCKET_NAME"],
+				BucketRegion:       cmData["BUCKET_REGION"],
+			}
+
+			// when
+			result, err := claimer.GetExternalStorageConfig(context.TODO(), device)
+
+			// then
+			Expect(err).To(BeNil())
+			Expect(result).ToNot(BeNil())
+			Expect(*result).To(Equal(expected))
```

Comment:
I think that there's no need. This is a minor issue of personal preference.
IMPO it is more accurate to use the object since this is what we are comparing and not the pointers.